pkg/util/fileutil/filesystem.go

```diff
@@ -42,3 +43,13 @@ func GetMountedFiles(targetPath string) (map[string]string, error) {
 	}
 	return paths, nil
 }
+
+// GetPodUIDFromTargetPath returns podUID from targetPath
+func GetPodUIDFromTargetPath(targetPath string) string {
+	re := regexp.MustCompile(`[\\|\/]+pods[\\|\/]+(.+?)[\\|\/]+volumes`)
```

Comment:
We should check and match on the whole volume path mounted by the driver
`volumes/kubernetes.io~secret` to prevent any override of other volumes that are mounted.

Reply:
Threw up this branch with the strict regex check - https://github.com/kubernetes-sigs/secrets-store-csi-driver/compare/master...aramase:regex?expand=1

We can use the `getVolumeFromTargetPath` to check if the volume name matches the volume name in the pod.spec.Volume[*] for the particular SPC.

Reply:
Okay I think I've got the regex fixed for this

Reply:
re-opening to look address aramase@ comment

Reply:
This took a bit to figure out the tests. I could add to the controller and reconciler.